### ISSUE TITLE
Allow option for Conditions to not be evaluated at compile time

### ISSUE THREAD

**Issue by smithhannahm** [awslabs/goformation#221]
If I am writing a template with conditions, on compile Goformation automatically processes the condition into a true/false output. For example, this:

```go
template.Parameters["Environment"] = &parameter{
	Type:          "String",
	Description:   "The type of environment the VPC will host",
	Default:       "dev",
	AllowedValues: []string{"dev", "prod"},
}
template.Conditions["IsProduction"] = cloudformation.Equals(cloudformation.Ref("Environment"), "prod")
```
produces:

```json
  "Conditions": {
    "IsProduction": false
  },
  "Parameters": {
    "Environment": {
      "AllowedValues": [
        "dev",
        "prod"
      ],
      "Default": "dev",
      "Description": "The type of environment the VPC will host",
      "Type": "String"
    },
```
While I would like it to produce this:
```json
  "Conditions": {
    "IsProduction": {
      "Fn::Equals": [
        {"Ref": "Environment"},
        "prod"
      ]
  },
  "Parameters": {
    "Environment": {
      "AllowedValues": [
        "dev",
        "prod"
      ],
      "Default": "dev",
      "Description": "The type of environment the VPC will host",
      "Type": "String"
    },
```

**Comment by JerzyLa** [comment]
@smithhannahm  Can you make Pull Request with this commit? I have tested it in my project and works fine.

**Comment by PaulMaddox** [comment]
Included in release v4.3.0
